scheduler/reconciler/reconcile_node.go

```diff
@@ -183,6 +215,27 @@ func (nr *NodeReconciler) diffSystemAllocsForNode(
 			continue
 		}
 
+		// find nodes on which to deploy canaries
+		canaryNode := slices.ContainsFunc(
+			canaryNodes[tg.Name],
+			func(n *structs.Node) bool { return n.ID == alloc.NodeID },
+		)
+
+		if canaryNode {
+			// we only deal with canaries if there's a deployment
+			if nr.DeploymentCurrent == nil {
+				continue
+			}
```

Comment:
By continuing here we're assuming that the node isn't also tainted and doesn't have reconnects. Is that true? Should we be instead guarding the `canaryNode` check with this `nr.DeploymentCurrent`?